vcx/dummy-cloud-agent/src/actors/agent.rs

```diff
@@ -423,6 +449,61 @@ impl Agent {
             .into_box()
     }
 
+    fn handle_update_configs(&mut self, msg: UpdateConfigs) -> ResponseActFuture<Self, Vec<A2AMessage>, Error> {
+
+        for config_option in msg.configs {
+            match config_option.name.as_str() {
+                "name" | "logo_url" => { *self.configs.entry(config_option.name.clone()).or_insert(config_option) = config_option.clone(); },
+                _ => continue
+            }
+        }
+
+        future::ok(())
+            .into_actor(self)
+            .and_then( |_, slf, _| {
+
+                let config_metadata: serde_json::Value = slf.configs.iter().map( |(_, v)| serde_json::to_value(v).unwrap() ).collect();
+
+                did::set_did_metadata(slf.wallet_handle, &slf.did, config_metadata.to_string().as_str())
+                    .map_err(|err| err.context("Can't store config data as DID metadata.").into())
+                    .into_actor(slf)
+            })
+            .map( |_, _, _| {
+                vec![A2AMessage::ConfigsUpdated( ConfigsUpdated {} )]
+            })
+            .into_box()
+    }
+
+    fn handle_get_configs(&mut self, msg: GetConfigs) -> ResponseActFuture<Self, Vec<A2AMessage>, Error> {
+
+        let configs: Vec<ConfigOption> = self.configs.iter().filter( |(k, _)| msg.configs.contains(k) ).map( |(_, v)| v.clone() ).collect();
+
+        future::ok(())
+            .into_actor(self)
+            .map( |_, _, _| { vec![A2AMessage::Configs( Configs { configs } )] })
+            .into_box()
+    }
+
+    fn handle_remove_configs(&mut self, msg: RemoveConfigs) -> ResponseActFuture<Self, Vec<A2AMessage>, Error> {
+
+        self.configs.retain( |k, _| !msg.configs.contains(k) );
+
+        future::ok(())
+            .into_actor(self)
+            .and_then( |_, slf, _| {
+
+                let config_metadata: serde_json::Value = slf.configs.iter().map( |(_, v)| serde_json::to_value(v).unwrap() ).collect();
+
```

Comment:
move outside and avoid mapping
`let config_metadata = ftry_act!(self, serde_json::to_string(&self.configs));`